drivers/platform/maxim/maxim_stdio.c

```diff
@@ -45,89 +45,94 @@
 #include <stdio.h>
 #include "maxim_stdio.h"
 
-#include "mxc_config.h"
 #include "mxc_sys.h"
-#include "board.h"
 #include "no-os/uart.h"
 #include "uart.h"
 
-#define CONSOLE_UART	(1)
-#define MXC_UARTn   MXC_UART_GET_UART(CONSOLE_UART)
-#define UART_FIFO   MXC_UART_GET_FIFO(CONSOLE_UART)
-
 #include <unistd.h>
 #include <sys/stat.h>
 
 #define STDIN_FILENO	0   /**> Definition of stdin */
 #define STDOUT_FILENO   1   /**> Definition of stdout */
 #define STDERR_FILENO   2   /**> Definition of stderr */
 
-static struct uart_desc *mdesc;
+static struct uart_desc *guart = NULL;
 
 void maxim_uart_stdio(struct uart_desc *desc)
 {
 	if (!desc)
 		return;
-	mdesc = desc;
-}
+	guart = desc;
 
-int _open(const char *name, int flags, int mode)
```

Comment:
The "Works correctly." part in the commit description is not exactly needed. I would prefer it mention why the `_open` API is removed.